pycytominer/cyto_utils/DeepProfiler_processing.py

```diff
@@ -0,0 +1,160 @@
+        """
+        self.filenames = self.index_df.apply(
+            self.build_filename_from_index, axis="columns"
+        )
+        self.filenames = [
+            pathlib.PurePath(f"{self.profile_dir}/{x}") for x in self.filenames
+        ]
+
+    def build_filename_from_index(self, row):
+        plate = row["Metadata_Plate"]
+        well = row["Metadata_Well"]
+        site = row["Metadata_Site"]
+
+        filename = f"{plate}_{well}_{site}{self.file_extension}"
+        return filename
+
+    def extract_filename_metadata(self, npz_file, delimiter="_"):
+        """
+        Format: plate_well_site.npz
+        """
+        base_file = os.path.basename(npz_file).strip(".npz").split(delimiter)
+        site = base_file[-1]
+        well = base_file[-2]
+        plate = delimiter.join(base_file[:-2])
+
+        return {"site": site, "well": well, "plate": plate}
+
+    def setup_aggregate(self):
+        if not hasattr(self, "filenames"):
+            self.build_filenames()
+
+        self.file_aggregate = {}
+        for filename in self.filenames:
+            file_info = self.extract_filename_metadata(filename, self.file_delimiter)
+            file_key = file_info[self.aggregate_on]
+
+            if self.aggregate_on == "site":
+                file_key = (
+                    f"{file_info['plate']}_{file_info['well']}_{file_info['site']}"
+                )
+
+            if self.aggregate_on == "well":
+                file_key = f"{file_info['plate']}_{file_info['well']}"
+
+            if file_key in self.file_aggregate:
+                self.file_aggregate[file_key]["files"].append(filename)
+            else:
+                self.file_aggregate[file_key] = {}
+                self.file_aggregate[file_key]["files"] = [filename]
+
+            self.file_aggregate[file_key]["metadata"] = file_info
+
+    def aggregate_deep(self):
+        if not hasattr(self, "file_aggregate"):
+            self.setup_aggregate()
+
+        self.aggregated_profiles = []
+        self.aggregate_merge_col = f"Metadata_{self.aggregate_on.capitalize()}_Position"
+
+        for metadata_level in self.file_aggregate:
+            df = pd.concat(
+                [load_npz(x) for x in self.file_aggregate[metadata_level]["files"]]
+            )
+            meta_df = pd.DataFrame(
+                self.file_aggregate[metadata_level]["metadata"], index=[0]
+            )
+            meta_df.columns = [f"Metadata_{x.capitalize()}" for x in meta_df.columns]
+
+            if self.aggregate_on == "well":
+                meta_df = meta_df.drop("Metadata_Site", axis="columns")
+
+            features = df.columns.tolist()
+            df = df.assign(Metadata_Aggregate_On=self.aggregate_on)
+            df = aggregate(
+                population_df=df,
+                strata="Metadata_Aggregate_On",
+                features=features,
+                operation=self.aggregate_operation,
```

Comment:
Not sure if it is necessary but do we want to provide the functionality of aggregate only a subset of sites? If so, I guess we will have to add a filter to `setup_aggregate` and also change `__init__()` so that it can accept a list of fields, like in https://github.com/cytomining/pycytominer/blob/fdd0e0935ab5446f5c3710e5270c5234d8ef18f1/pycytominer/cyto_utils/cells.py#L52-L53 

We could also ignore this for the time being and add the functionality later.

Reply:
I will add to the list of additial things, not in this PR

Reply:
#144